edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/routing/RoutingEdgeHub.cs

```diff
@@ -214,16 +216,28 @@ internal async Task ProcessSubscription(string id, Option<ICloudProxy> cloudProx
             }
         }
 
-        async void CloudConnectionEstablished(object sender, IIdentity identity)
+        async void DeviceConnected(object sender, EventArgs eventArgs)
         {
+            Events.DeviceConnectedProcessingSubscriptions();
             try
             {
-                Events.ProcessingSubscriptions(identity);
-                await this.ProcessSubscriptions(identity.Id);
+                IEnumerable<IIdentity> connectedClients = this.connectionManager.GetConnectedClients();
```

Comment:
Should we `ToList` this to get a consistent snapshot? Is it possible for the `IEnumerable` from the `GetConnectedClients` to change while iterating?

Reply:
Emm, it could.. let me add ToList()